StRoot/StFwdTrackMaker/include/Tracker/FwdHit.h

```diff
@@ -0,0 +1,164 @@
+
+// small class to store Mc Track information
+class McTrack {
+  public:
+    McTrack() {
+        _pt = -999;
+        _eta = -999;
+        _phi = -999;
+        _q = 0;
+    }
+    McTrack(float pt, float eta = -999, float phi = -999, int q = 0,
+            int start_vertex = -1) {
+        _pt = pt;
+        _eta = eta;
+        _phi = phi;
+        _q = q;
+        _start_vertex = start_vertex;
+    }
+    void addHit(KiTrack::IHit *hit) { hits.push_back(hit); }
+
+    void addFSIHit(KiTrack::IHit *hit) { fsi_hits.push_back(hit); }
+
+    float _pt, _eta, _phi;
+    int _tid, _q, _start_vertex;
+
+    std::vector<KiTrack::IHit *> hits;
+    std::vector<KiTrack::IHit *> fsi_hits;
+};
+
+class FwdHit : public KiTrack::IHit {
+  public:
+    FwdHit(unsigned int id, float x, float y, float z, int vid, int tid,
+           TMatrixDSym covmat, std::shared_ptr<McTrack> mcTrack = nullptr )
+        : KiTrack::IHit() {
+        _id = id;
+        _x = x;
+        _y = y;
+        _z = z;
+        _tid = tid;
+        _vid = vid;
+        _mcTrack = mcTrack;
+        _hit = 0;
+        _covmat.ResizeTo( 3, 3 );
+        _covmat = covmat;
+
+        int _map[] = {0, 0, 0, 0, 0, 1, 2, 0, 0, 3, 4, 5, 6}; // ftsref6a
+
+        if (vid > 0)
+            _sector = _map[vid];
+        else {
+            _sector = abs(vid); // set directly if you want
+            // now set vid back so we retain info on the tru origin of the hit
+            _vid = abs(vid) + 9; // we only use this for sTGC only.  Needs to be
+                                 // cleaner in future.
+        }
+    };
+
+    const KiTrack::ISectorSystem *getSectorSystem() const {
+        return gFwdSystem;
+    } // need to implement
+
+    int _tid; // aka ID truth
+    int _vid;
+    unsigned int _id; // just a unique id for each hit in this event.
+    std::shared_ptr<McTrack> _mcTrack;
+    TMatrixDSym _covmat;
+
+    StHit *_hit;
+};
+
+using Seed_t = std::vector<KiTrack::IHit *>;
+
+class FwdConnector : public KiTrack::ISectorConnector {
+  public:
+    FwdConnector(unsigned int distance)
+        : _system(*gFwdSystem), _distance(distance) {}
+    ~FwdConnector(){/**/};
+
+    // Return the possible sectors (layers) given current
+    virtual std::set<int> getTargetSectors(int disk) {
+
+        std::set<int> r;
+
+        if (disk > 0 && _distance >= 1)
+            r.insert(disk - 1);
+
+        if (disk > 1 && _distance >= 2)
+            r.insert(disk - 2);
+
+        if (disk > 2 && _distance >= 3)
+            r.insert(disk - 3);
+
+        if (disk > 3 && _distance >= 4)
+            r.insert(disk - 4);
+
+        return r;
+    };
+
+  private:
+  protected:
+    const FwdSystem _system; // numbering system
+    unsigned int _distance;  // number of layers forward to search
+};                           // FwdConnector
+
+class SeedQual {
+  public:
+    inline double operator()(Seed_t s) { return double(s.size()) / 7.0; }
+};
+
+class SeedCompare {
+  public:
```

Comment:
Why do you define everything as class and then give public access? Isn't cleaner to just define the types as struct in such cases?